_RelativeAppBundlePath fails due to _AppBundlePath being a full path

### Steps to Reproduce

1. Create a Maui IOS project in Visual Studio 2022 Version 17.3.0 Preview 1.1
2. Modify Directory.Build.props to set output paths to something other than defaults.  Below is an example.
```
    <RepoRootPath>$(MSBuildThisFileDirectory)</RepoRootPath>
    <BaseIntermediateOutputPath>$(RepoRootPath)obj\$([MSBuild]::MakeRelative($(RepoRootPath), $(MSBuildProjectDirectory)))\</BaseIntermediateOutputPath>
    <BaseOutputPath Condition=" '$(BaseOutputPath)' == '' ">$(RepoRootPath)bin\$(MSBuildProjectName)\</BaseOutputPath>
    <PackageOutputPath>$(RepoRootPath)bin\Packages\$(Configuration)\</PackageOutputPath>
```
3. Build

### Expected Behavior
_RelativeAppBundlePath to handle an _AppBundlePath that is fully qualified.  

### Actual Behavior
```
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.402\targets\Xamarin.Shared.Sdk.targets, line 890

Error	MSB4184	The expression "[MSBuild]::MakeRelative(D:\GitHub\NetworkVisor\Platform\bin\NetworkVisor.Platform.Test.Devices.IOS\Debug\net6.0-ios\iossimulator-x64\publish\, D:\GitHub\NetworkVisor\Platform\test\ios\NetworkVisor.Platform.Test.Devices.IOS/D:\GitHub\NetworkVisor\Platform\bin\NetworkVisor.Platform.Test.Devices.IOS\Debug\net6.0-ios\iossimulator-x64\NetworkVisor.Platform.Test.Devices.IOS.app\)" cannot be evaluated. The given path's format is not supported.	NetworkVisor.Platform.Test.Devices.IOS	C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.402\targets\Xamarin.Shared.Sdk.targets	890	
```

### Environment

<details>
<summary>Version information</summary>
```
Microsoft Visual Studio Enterprise 2022
Version 17.3.0 Preview 1.1
VisualStudio.17.Preview/17.3.0-pre.1.1+32519.111
Microsoft .NET Framework
Version 4.8.04161

Installed Version: Enterprise

Architecture Diagrams and Analysis Tools   00476-80000-00000-AA426
Microsoft Architecture Diagrams and Analysis Tools

Visual C++ 2022   00476-80000-00000-AA426
Microsoft Visual C++ 2022

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   17.3.122.33185
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2012   17.3.122.33185
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.3.122.33185
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.3.122.33185
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.3.0-1.22254.1+9919d7e7bd753404a5d2328e5e3fb2de635169f3
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   17.0.22115.6
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

GhostDoc   22.1.22090.0  (Not Loaded)
Generate XML Comments from your code, maintain clean and up-to-date documentation, produce help documentation in multiple formats, use intelligent source code Spell Checker in Visual Studio.

Linux Core Dump Debugging   1.0.9.32505
Enables debugging of Linux core dumps.

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.3.11 (4f73250)
Support for debugging Mono processes with Visual Studio.

Node.js Tools   1.5.40427.1 Commit Hash:d6dd02cc1b56ad1cb24b4ef0c48a81a985bbc64c
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   6.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Office Developer Tools for Visual Studio   17.0.32314.00
Microsoft Office Developer Tools for Visual Studio

Python - Django support   17.0.22115.6
Provides templates and integration for the Django web framework.

Python - Profiling support   17.0.22115.6
Profiling support for Python projects.

Python with Pylance   17.0.22115.6
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Razor (ASP.NET Core)   17.0.0.2222701+751db1ebea5e6a9ecc7fa57fe447180422afa610
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62204.01010
Microsoft SQL Server Data Tools

Syntax Visualizer   1.0
An extension for visualizing Roslyn SyntaxTrees.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.10420.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.3.0-1.22254.1+9919d7e7bd753404a5d2328e5e3fb2de635169f3
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual C++ for Cross Platform Mobile Development (Android)   17.0.32505.426
Visual C++ for Cross Platform Mobile Development (Android)

Visual C++ for Cross Platform Mobile Development (iOS)   17.0.32505.426
Visual C++ for Cross Platform Mobile Development (iOS)

Visual C++ for Linux Development   1.0.9.32505
Visual C++ for Linux Development

Visual F# Tools   17.1.0-beta.22178.3+6da0245a7ce4bb8483b8d1f2993c8ecaea967ad9
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Workflow Manager Tools 1.0   1.0
This package contains the necessary Visual Studio integration components for Workflow Manager.

Xamarin   17.3.0.151 (d17-3-p1.1@5a59bfa)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.3.0.33 (remotes/origin/d17-3-1@9c0d11a29)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.2.15 (2e3b60e)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   12.3.99.58 (main/64b22fc)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: adf1bc4
    Java.Interop: xamarin/java.interop/release/6.0.3xx@2a882d2d
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.38.2@7b1e016
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@fc3c2ac


Xamarin.iOS and Xamarin.Mac SDK   15.11.0.183 (f8314a435)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```
</details>

18:08:41.562   1:2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.402\targets\Xamarin.Shared.Sdk.targets(890,4): error MSB4184: The expression "[MSBuild]::MakeRelative(D:\GitHub\NetworkVisor\Platform\bin\NetworkVisor.Platform.Test.Devices.IOS\Debug\net6.0-ios\ios-arm64\publish\, D:\GitHub\NetworkVisor\Platform\test\ios\NetworkVisor.Platform.Test.Devices.IOS/D:\GitHub\NetworkVisor\Platform\bin\NetworkVisor.Platform.Test.Devices.IOS\Debug\net6.0-ios\ios-arm64\device-builds\iphone14.3-15.4.1/NetworkVisor.Platform.Test.Devices.IOS.app\)" cannot be evaluated. The given path's format is not supported. [D:\GitHub\NetworkVisor\Platform\test\ios\NetworkVisor.Platform.Test.Devices.IOS\NetworkVisor.Platform.Test.Devices.IOS.csproj]

--- a/msbuild/Xamarin.Shared/Xamarin.Shared.targets
+++ b/msbuild/Xamarin.Shared/Xamarin.Shared.targets
@@ -2683,7 +2683,10 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 			<AppBundleDir>$(_AppContainerDir)$(_AppBundleName)$(AppBundleExtension)</AppBundleDir>
 		</PropertyGroup>
 		<PropertyGroup>
-			<_AppBundlePath>$(AppBundleDir)\</_AppBundlePath>
+			<!-- Ensure _AppBundlePath is a relative path (relative to the project directory) and contains a trailing slash -->
+			<_AppBundlePath>$(AppBundleDir)</_AppBundlePath>
+			<_AppBundlePath Condition="$([System.IO.Path]::IsPathRooted('$(AppBundleDir)'))">$([MSBuild]::MakeRelative('$(MSBuildProjectDirectory)','$(AppBundleDir)'))</_AppBundlePath>
+			<_AppBundlePath>$([MSBuild]::EnsureTrailingSlash('$(_AppBundlePath)'))</_AppBundlePath>
 			<_AppResourcesRelativePath Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">Contents\Resources\</_AppResourcesRelativePath>
 			<_AppResourcesRelativePath Condition="'$(_PlatformName)' == 'iOS' Or '$(_PlatformName)' == 'tvOS' Or '$(_PlatformName)' == 'watchOS'"></_AppResourcesRelativePath>
 			<_AppResourcesPath>$(_AppBundlePath)$(_AppResourcesRelativePath)</_AppResourcesPath>

